### ISSUE TITLE
didInitAttrs cannot see bound values unless they are explicitly defined on the component class.

### ISSUE THREAD

**Issue by jakesjews** [emberjs/ember.js#12035]
I just noticed this in 1.13.7. If a property isn't explicitly defined on a component's class then its value is never seen in the didInitAttrs method even though it is bound in the template. Defining the property with a value of null does work but this wasn't required in 1.13.6. I'm not 100% sure if this is a bug or if the previous behavior was incorrect and it's been fixed now.

Ember-twiddle at http://ember-twiddle.com/62d39c8e4f57e0eb1fbb

**Comment by aortbals** [comment]
I can confirm this bug. Immediately after upgrading, I am seeing `undefined` for an attribute passed to a component in `init` of the component:

```handlebars
{{foo-component bar='something'}}
```

```js
init() {
  Ember.assert("The {{foo-component}} attribute 'bar' was not configured", this.get('bar'));
},
```

The following assert throws in 1.13.7 where it would be correctly bound to `'something'` in 1.13.6. I can also confirm that defining the attribute on the component as `null` fixes the issue. In my experience, it is pretty common to see attributes passed to the component that aren't explicitly defined (regardless of what people feel about the practice of non-explicit attributes).

**Comment by martndemus** [comment]
I bumped into this too. This was during `didReceiveAttrs`.
If you declare `foo: null` on the object things work again.

**Comment by EricSchank** [comment]
I think I have a fix for this, but I'm writing a test to verify -- unless one of you has a breaking test already?

**Comment by aortbals** [comment]
I do not. That's great to hear though!

**Comment by noslouch** [comment]
don't have a breaking test, just a breaking app :-/

**Comment by gerry3** [comment]
+1

**Comment by rwjblue** [comment]
Fixed by #12071.

**Comment by rwjblue** [comment]
Will be in a 1.13.8 soonish.
